File: rank_test_problems_b_0002/rank_test_problems_b_0002__4.py
# 実行時間 10ms
n, m = list(map(int, input().split()))
g = [[0 for _ in [0] * m] for _ in [0] * n]
q = int(input())
z = 0

for _ in [0] * q:
    p, x, y = input().split()
    x = int(x) - 1
    y = int(y) - 1

    if -2 < g[x][y] < 2:
        if p == "A":
            g[x][y] += 1
        elif p == "B":
            g[x][y] -= 1
for i in g:
    for j in i:
        if 1 <= j:
            z += 1
        elif j <= -1:
            z -= 1
print("A" if 0 < z else "B" if z < 0 else "Draw")

File: rank_test_problems_b_0002/rank_test_problems_b_0002__4.cs
// 実行時間 20ms
using static System.Console;
using static System.Linq.Enumerable;

internal class Program
{
    static void Main()
    {
        int[] nm = ReadLine().Split().Select( int.Parse ).ToArray();
        int[][] g = new int[nm[0]][];
        int q = int.Parse( ReadLine() );
        int z = 0;

        foreach( int i in Range( 0, nm[0] ) ) {
            g[i] = new int[nm[1]];
        }
        foreach( int _ in Range( 0, q ) ) {
            string[] s = ReadLine().Split();
            int x = int.Parse( s[1] ) - 1;
            int y = int.Parse( s[2] ) - 1;

            if( 2 <= g[x][y] || g[x][y] <= -2 ) {
                continue;
            } else if( s[0][0] == 'A' ) {
                g[x][y]++;
            } else if( s[0][0] == 'B' ) {
                g[x][y]--;
            }
        }
        foreach( int[] i in g ) {
            foreach( int j in i ) {
                if( 1 <= j ) {
                    z++;
                } else if( j <= -1 ) {
                    z--;
                }
            }
        }
        WriteLine( 0 < z ? "A" : z < 0 ? "B" : "Draw" );
        return;
    }
}
